guides/release/testing/testing-components.md

```diff
@@ -276,7 +276,19 @@ module('Integration | Component | comment-form', function(hooks) {
 
 In cases where components have dependencies on Ember services,
 it is possible to stub these dependencies for rendering tests.
-You stub Ember services by using the built-in `register()` function to register your stub service in place of the default.
+You stub non-Ember services by using the built-in `register()` function to register your stub service in place of the default.
+
+<div class="cta">
+  <div class="cta-note">
+    <div class="cta-note-body">
+      <div class="cta-note-heading">Zoey says...</div>
+      <div class="cta-note-message">
+        Not all services can or should be stubbed. Certain services like router should never be stubbed. If an integration or unit test relies on routing then that behavior would be better tested as an acceptance test. The behavior of internal services like router may change over time and you should not attempt to override their behavior.
```

Comment:
🤔 

I don't know if I agree with the statement about using an acceptance test. 

You can setup the router in unit and integration tests. This probs also related to the recent feature of lazily setting up the router upon access.
The main gotchya is that the router needs a map and a transition before it'll have useful behavior